File: Data Structures/Stack.py
# -*- coding: utf-8 -*-
"""
Created on Sun Sep 23 10:08:43 2018

@author: swats
"""

class Stack():
    def __init__(self):
        self.items=[]
        
    def push(self,item):
        self.items.append(item)
        
    def pop(self):
        self.items.pop()
    
    #top of the stack
    def peek(self):
        if not self.is_empty():
            return self.items[-1]
    
    def is_empty(self):
        return self.items == []
    
    def display(self):
        return self.items
    
s=Stack()
'''
#print(s.is_empty())
s.push("A")
s.push("B")
s.push("C")
s.push("D")
'''
#print(s.display())
#print(s.is_empty())
#print(s.peek())
#s.pop()
#print(s.display())
#print(s.peek())

File: Java/Stack.java

public class Stack
{
    final int n=10;
    int a[]=new int [n];
    int top;

    public static void main(String[] args)
    {
        Stack s=new Stack();
        s.push(5);
        s.push(6);
        s.push(7);
        s.pop();

    }

    Stack()
    {
        top=-1;
    }

    public void push(int data)
    {
        if(top>(n-1))
        {
            System.out.println("Stack full");
        }
        else
        {
            top++;
            a[top]=data;
            System.out.println(data+" pushed into stack");
        }
    }

    public void pop()
    {
        if(top<0)
        {
            System.out.println("Stack underflow");
        }
        else
        {
            System.out.println(a[top]+" popped from stack");
            a[top]=a[top--];

        }
    }

    public void display()
    {
        for(int i=0;i<=n-1;i++)
            System.out.println(a[i]);

    }

}
